mlflow/pytorch/_pytorch_autolog.py

```diff
@@ -108,12 +108,25 @@ def _log_metrics(self, trainer, pl_module):
                     if isinstance(callback, pl.callbacks.early_stopping.EarlyStopping):
                         self._early_stop_check(callback)
 
+            _pl_version = Version(pl.__version__)
+
+            # In pytorch-lightning >= 1.4.0, validation is run inside the training epoch and
+            # `trainer.callback_metrics` contains both training and validation metrics of the
+            # current training epoch when `on_train_epoch_end` is called:
+            # https://github.com/PyTorchLightning/pytorch-lightning/pull/7357
+            if _pl_version >= Version("1.4.0dev"):
```

Comment:
The current dev version of `pytorch-lightning` is `1.4.0dev`. To enable the fix in this version, use `1.4.0dev` instead of `1.4.0`.